os/mm/kmm_heap/kmm_realloc.c

```diff
@@ -144,21 +144,26 @@ FAR void *kmm_realloc(FAR void *oldmem, size_t newsize)
 	mmaddress_t caller_retaddr = 0;
 	ARCH_GET_RET_ADDRESS(caller_retaddr)
 #endif
-	struct mm_heap_s *kheap_origin = mm_get_heap(oldmem);
+	struct mm_heap_s *kheap_origin;
 	struct mm_heap_s *kheap_new;
 
-	if (newsize == 0) {
-		mm_free(kheap_origin, oldmem);
-		return NULL;
-	}
+	if (oldmem) {
+		kheap_origin = mm_get_heap(oldmem);
+		ASSERT(kheap_origin);
```

Comment:
Let's leave some explanation about this. How about below?
```
// The oldmem given by first argument is not a dynamically allocated address. This will cause ASSERT like Linux.
```